src/supertux/game_object_manager.hpp

```diff
@@ -116,10 +116,11 @@ class GameObjectManager
   template<class T>
   T& get_singleton_by_type() const
   {
-    const auto& range = get_objects_by_type<T>();
-    assert(range.begin() != range.end());
-    assert(range.begin()->is_singleton());
-    return *range.begin();
+    const auto& objs = get_objects_by_type_index(typeid(T));
+    assert(objs.size() == 1);
+    T* obj = dynamic_cast<T*>(objs[0]);
+    assert(obj);
+    return *obj;
```

Comment:
Since we got the object via a typeindex, we know it's of that type, so a `static_cast` here would be enough.

Reply:
Done.